controllers/keyword.go

```diff
@@ -32,44 +32,65 @@ func (k *KeywordController) applyRoutes(engine *gin.RouterGroup) {
 }
 
 func (k *KeywordController) displayKeyword(c *gin.Context) {
-	html.RenderWithFlash(c, http.StatusOK, keywordView, keywordTitle, nil)
+	keywordService := initKeywordService(c)
+	data := getKeywordsData(keywordService)
+
+	html.RenderWithFlash(c, http.StatusOK, keywordView, keywordTitle, data)
 }
 
 func (k *KeywordController) uploadKeyword(c *gin.Context) {
+	keywordService := initKeywordService(c)
+	data := getKeywordsData(keywordService)
+
 	form := &UploadFileForm{}
 	if err := c.ShouldBind(form); err != nil {
 		for _, fieldErr := range err.(validator.ValidationErrors) {
-			html.RenderWithError(c, http.StatusBadRequest, keywordView, keywordTitle, errorHandler.ValidationErrorMessage(fieldErr), nil)
+			html.RenderWithError(c, http.StatusBadRequest, keywordView, keywordTitle, errorHandler.ValidationErrorMessage(fieldErr), data)
 			return
 		}
 	}
 
 	// Validate if file is CSV type
-	if err := models.ValidateFileType(form.File.Header["Content-Type"][0]); err != nil {
-		html.RenderWithError(c, http.StatusBadRequest, keywordView, keywordTitle, err, nil)
+	validateTypeErr := keywordService.ValidateFileType(form.File.Header["Content-Type"][0])
+	if validateTypeErr != nil {
+		html.RenderWithError(c, http.StatusBadRequest, keywordView, keywordTitle, validateTypeErr, data)
 		return
 	}
 
-	filename := models.UploadFile(c, form.File)
-	record, readFileErr := models.ReadFile(filename)
+	filename := keywordService.UploadFile(c, form.File)
+
+	record, readFileErr := keywordService.ReadFile(filename)
```

Comment:
Naming this `record` is a bit confusing now with hindsight as it's a slice of strings 🤔  Even in the method `ReadFile`, using `record` is misleading. I would not advise using the plural `records` as plurals, in general, should be avoided to avoid typos. How about `wordList` (not to confuse with `keywordList` since these are raw words arsed from teh CSV file) or `parsedKeywordList`?

Reply:
Agree with you. Fixed in 13b2e6b4c493d3dc0f35d548b634c49daa4ebb74